apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1.IntegrationTests/Emulator/metadata_server.py

```diff
@@ -0,0 +1,366 @@
+    return ''.join(unicode(i) + u'/\n' for i in range(len(self._items)))
+
+class MetadataJSONEncoder(json.JSONEncoder):
+  def ensureOrderedDict(self, o):
+    if isinstance(o, OrderedDict):
+      return self.fixupOrderedDict(o)
+
+    return o
+
+  def fixupOrderedDict(self, d):
+    result = OrderedDict()
+    for key in d:
+      if excludeKeyInRecursiveGet(key):
+        continue
+
+      if key.endswith(u'@developer.gserviceaccount.com'):
+        newKey = key
+      else:
+        newKey = common.convertToCamelCase(key)
+
+      if (d[key] == numeric_project_id_token):
+        result[newKey] = numeric_project_id
+      else:
+        result[newKey] = self.default(d[key])
+
+    return result
+
+  def encode(self, o):
+    return json.JSONEncoder.encode(self, self.ensureOrderedDict(o))
+
+  def default(self, o):
+    if isinstance(o, IndexedList):
+      return [self.default(item) for item in o]
+    
+    return self.ensureOrderedDict(o)
+
+not_found_error = formatResponse(u'Error: Not found', 404, common.content_type_html)
+bad_request_error = formatResponse(u'Error: bad request', 400, common.content_type_html)
+
+all_data = {}
+def loadData():
+  global all_data
+  with open(os.path.join(os.path.dirname(os.path.realpath(__file__)), 'initial_data.json')) as initial_data_file:
+    all_data = json.load(initial_data_file, object_pairs_hook=OrderedDict)
+    indexed_lists = [
+      u'instance/disks',
+      u'instance/licenses',
+      u'instance/network-interfaces',
+      u'instance/network-interfaces/*/access-configs',
+      u'instance/network-interfaces/*/forwarded-ips',
+      u'instance/network-interfaces/*/ip-aliases'
+    ]
+    for indexed_list in indexed_lists:
+      components = indexed_list.split('/')
+      def process(data, i):
+        component = components[i]
+        if component == '*':
+          for item in data:
+            process(item, i + 1)
+        elif i == len(components) - 1:
+          data[component] = IndexedList(data[component])
+        else:
+          process(data[component], i + 1)
+
+      process(all_data, 0)
+
+def getRawDataAt(path):
+  result = all_data
+  for component in path.split('/'):
+    if len(component) == 0:
+      continue
+
+    if isinstance(result, IndexedList):
+      try:
+        index = int(component)
+      except ValueError:
+        return (None, not_found_error)
+      
+      if 0 <= index and index < len(result):
+        result = result[index]
+      else:
+        return (None, not_found_error)
+    else:
+      component = component.replace(unicode(numeric_project_id), numeric_project_id_token)
+      component = component.replace(project_id, project_id_token)
+      if not component in result:
+        return (None, not_found_error)
+
+      result = result[component]
+
+  if result == numeric_project_id_token:
+    result = numeric_project_id
+
+  return (result, None)
+
+def convertToTextRecursive(prefix, raw_data):
+  text = u''
+  if isinstance(raw_data, list):
```

Comment:
I had more comments but github ate them -_- but many of them centered around a theme, which was: why do we need to do all this type checking? it seems like a smell...generally we should convert to the data types that we want, and deal with them. this is especially true when we control the data end to end